src/robotFactory.js

```diff
@@ -1,17 +1,85 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+
+  return this;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+
+  return this;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+
+  return this;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+
+  return this;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  const { name, chipVersion, weight } = this;
+
+  return `Robot: ${name}, Chip version: ${chipVersion}, Weight: ${weight}`;
+};
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+  this.coords.z = 0;
 }
 
-function DeliveryDrone() {
-  // implement
+FlyingRobot.prototype = BaseRobot.prototype;
+
+FlyingRobot.prototype.goUp = function(step = 1) {
+  this.coords.z += step;
+
+  return this;
+};
+
+FlyingRobot.prototype.goDown = function(step = 1) {
+  this.coords.z -= step;
+
+  return this;
+};
+
+function DeliveryDrone(name, weight, coords,
+  chipVerion, maxLoadWeight, currentLoad) {
```

Comment:
```suggestion
function DeliveryDrone(
  name, 
  weight, 
  coords,
  chipVerion, 
  maxLoadWeight, 
  currentLoad
) {
```
one line for one parametr